scripts/environment/gsutil_env.py

```diff
@@ -0,0 +1,281 @@
+                        self.policy_bucket,
+                        self.output_bucket))
+
+    def ensure_environment(self):
+        """
+        Check the following:
+        1. gsutil tool is installed
+        2. gcloud tool/environment is logged in (gcloud auth login)
+        3. A project_id has been set on the environment
+        """
+        gsutil_cmd = find_executable('gsutil')
+        if gsutil_cmd:
+            print 'Found gsutil tool!'
+        else:
+            raise EnvironmentError(
+                'Could not find gsutil. '
+                'Have you installed the Google Cloud SDK?\n'
+                'You can get it here: https://cloud.google.com/sdk/')
+        p = Popen(['gcloud', 'info', '--format=json'], stdout=PIPE, stderr=PIPE)
+        output, err = p.communicate()
+        try:
+            gcloud_info = json.JSONDecoder().decode(output)
+            if 'config' in gcloud_info:
+                self.gcloud_config.auth_account = gcloud_info['config']['account']
+                self.gcloud_config.project_id = gcloud_info['config']['project']
+        except ValueError:
+            print 'Failed to get gcloud info, will initialize gcloud'
+
+        if not self.gcloud_config.auth_account:
+            self.gcloud_config.auth_login()
+
+        if not self.gcloud_config.project_id:
+            self.gcloud_config.create_or_use_project()
+
+        print self.gcloud_config
+
+    def choose_bucket(self, object_type):
+        """
+        Create or use a specified GCS bucket to store the specified object.
+        Also turn on object versioning on the bucket if it stores policy
+        objects.
+        """
+        p = Popen(['gsutil', 'ls'], stdout=PIPE, stderr=PIPE)
+        output, err = p.communicate()
+        if p.returncode:
+            print err
+            from sys import exit
+            exit(p.returncode)
+
+        prompt_templ = ('+----------------------+\n'
+                        '| Set up {} bucket |\n'
+                        '+----------------------+\n'
+                        'Specify a bucket for {}:')
+        prompt_header = ''
+        if object_type == ScannerBucketObject.POLICY:
+            prompt_header = prompt_templ.format('POLICY', 'your policies')
+        elif object_type == ScannerBucketObject.OUTPUT:
+            prompt_header = prompt_templ.format('OUTPUT', 'the scanner output')
+
+        while True:
+            bucket_choice = raw_input(
+                ('{}\n'
+                 '[1] Enter bucket name\n'
+                 '[2] Choose from list\n'
+                 'Enter your numeric choice: '
+                 .format(prompt_header))).strip()
+
+            bucket = None
+            if bucket_choice == '1':
+                bucket = self._create_or_reuse_bucket(object_type)
+            elif bucket_choice == '2':
+                bucket = self._choose_bucket(output)
+            else:
+                continue
+
+            if bucket:
```

Comment:
is this needed?  seems it will always be "1" or "2" at this point from the use of continue above